internal/service/cognitoidp/user_pool.go

```diff
@@ -463,6 +463,11 @@ func ResourceUserPool() *schema.Resource {
 							Required:     true,
 							ValidateFunc: verify.ValidARN,
 						},
+						"sns_region": {
+							Type:         schema.TypeString,
+							Optional:     true,
```

Comment:
```suggestion
							Optional:     true,
							Computed:     true,
```
This is needed because the AWS API evidently defaults this to the current region. Without it the existing tests fail with a spurious diff, trying to change `sns_region = "us-west-2" -> null`